include/dungeon_entity.h

```diff
@@ -1,44 +1,227 @@
 #ifndef GUARD_DUNGEON_ENTITY_H
 #define GUARD_DUNGEON_ENTITY_H
 
-struct subStruct_8048524
+#include "item.h"
+#include "move.h"
+
+struct DungeonEntityData
 {
-    /* 0x0 */ u8 unk0;
-    /* 0x1 */ u8 unk1;
-    /* 0x2 */ s16 species;
-    /* 0x4 */ s16 transformSpecies;
+    // This has different purposes for Pokémon, items, and traps.
+    // Pokemon: MovementFlag
+    // Items: ItemFlag
+    // Traps: TrapType
+    /* 0x0 */ u8 flags;
+    /* 0x1 */ bool8 moving;
+    /* 0x2 */ u16 entityType; // Pokémon species or item ID.
+    // Everything from here on only applies to Pokémon.
+    /* 0x4 */ u16 transformSpecies; // Shows a different Pokémon when using Transform.
     /* 0x6 */ bool8 isEnemy;
     /* 0x7 */ bool8 isLeader;
-    /* 0x8 */ u8 shopKeeper;
+    /* 0x8 */ u8 shopkeeperMode;
     /* 0x9 */ u8 level;
-    /* 0xA */ u16 partyIndex;
-    /* 0xC */ u16 IQ;
-    /* 0xE */ s16 HP;
-    /* 0x10 */ s16 maxHP;
-    /* 0x12 */ s16 bossHPBackup;
-    u8 fill14[0x46 - 0x14];
+    /* 0xA */ u8 partyIndex; // Leader is 0, partner is 1, etc.
+    u8 fillB;
+    /* 0xC */ u16 iq;
```

Comment:
```suggestion
    /* 0xC */ u16 IQ;
```

Reply:
Is it convention here to capitalize acronyms even at the start of a variable name? I'm used to starting all variable names with lowercase.

Reply:
Given it's an abbreviation, I've been keeping those in uppercase. Otherwise, I've been doing camel-case typically.

Reply:
Okay, should something like `iqSkillsSelected` also be renamed to `IQSkillsEnabled`?

Reply:
Yeah, sounds good

Reply:
Renamed HP/IQ vars accordingly.